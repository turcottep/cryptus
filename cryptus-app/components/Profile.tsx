import React from "react";
import { signIn, signOut, useSession } from "next-auth/client";

export default function NavbarLandingPage() {
  const [session, loading] = useSession();

  return (
    <div id="header" className=" w-full top-0 text-black bg-instagram">
      <div className="w-full flex flex-col justify-center h-full md:py-4">
        <div className="flex w-full justify-between items-center md:items-end py-4">
          <img
            className="self-start w-20 h-20 rounded-full border-black border-2 mx-2"
            src="./icons/icon-192x192.png"
            alt="icon"
          />
          <div className="flex w-full h-full justify-between flex-col">
            <span className="w-full text-left align-top text-gray-700 font-semibold text-2xl lg:text-2xl pb-1 pl-4">
              Alexandre Lafleur
            </span>
            <div className="w-full text-sm px-4 pb-2">
              is simply dummy text of the printing and typesetting industry.
              Lorem Ipsum has been the industry's standard dummy text ever since
              the
            </div>
          </div>
        </div>
        <div className="flex w-full justify-around flex-row text-center border-b border-t border-gray-400 ">
          <div className="flex-col ">
            <span className="font-bold">7</span>
            <br />
            <span className="text-gray-400">Owned</span>
          </div>
          <div className="flex-col">
            <span className="font-bold">3</span>
            <br />
            <span className="text-gray-400">On sale</span>
          </div>
          <div className="flex-col">
            <span className="font-bold">24</span>
            <br />
            <span className="text-gray-400">Sold</span>
          </div>
        </div>
        <div className="flex justify-around items-center py-2">
          <svg
            className="h-6 w-6"
            viewBox="-2 -2 22 22"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 4H4V0H0V4ZM6 16H10V12H6V16ZM0 16H4V12H0V16ZM0 10H4V6H0V10ZM6 10H10V6H6V10ZM12 0V4H16V0H12ZM6 4H10V0H6V4ZM12 10H16V6H12V10ZM12 16H16V12H12V16Z"
              fill="#999999"
            />
          </svg>

          <svg
            className="h-6 w-6 content-center my-auto"
            viewBox="0 0 24 24"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.41 11.58L12.41 2.58C12.05 2.22 11.55 2 11 2H4C2.9 2 2 2.9 2 4V11C2 11.55 2.22 12.05 2.59 12.42L11.59 21.42C11.95 21.78 12.45 22 13 22C13.55 22 14.05 21.78 14.41 21.41L21.41 14.41C21.78 14.05 22 13.55 22 13C22 12.45 21.77 11.94 21.41 11.58ZM13 20.01L4 11V4H11V3.99L20 12.99L13 20.01Z"
              fill="#999999"
            />
            <path
              d="M6.5 8C7.32843 8 8 7.32843 8 6.5C8 5.67157 7.32843 5 6.5 5C5.67157 5 5 5.67157 5 6.5C5 7.32843 5.67157 8 6.5 8Z"
              fill="#999999"
            />
          </svg>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6"
            fill="none"
            viewBox="0 0 24 24"
            stroke="#999999"
            id="heart-emoji"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
            />
          </svg>
        </div>
      </div>
    </div>
  );
}
