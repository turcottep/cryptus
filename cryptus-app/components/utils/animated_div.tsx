import React from "react";
// import { AnimatePresence, motion } from "framer-motion";

export default function AnimatedDiv(props) {
  return (
    // <AnimatePresence>
    //   <motion.div
    //     key="modal"
    //     initial={{ x: "-100vw", opacity: 0 }}
    //     animate={{ x: 0, opacity: 1 }}
    //     transition={{ type: "spring", bounce: 0.25, duration: 0.8 }}
    //     exit={{ opacity: 0 }}
    //   >
    //     {props.children}
    //   </motion.div>
    // </AnimatePresence>
    <div>{props.children}</div>
  );
}
