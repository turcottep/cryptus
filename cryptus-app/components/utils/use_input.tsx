import React, { useState } from "react";

type InputProps = {
  id: string;
  type: string;
  initialValue: any;
  placeholder: string;
};

export default function useInput(inputProps: InputProps) {
  const { id, placeholder, initialValue } = inputProps;

  const [value, setValue] = useState(initialValue);

  return {
    value,
    setValue,
    reset: () => setValue(""),
    bind: {
      value,
      id: id,
      placeholder: placeholder,
      onChange: (e) => {
        setValue(e.target.value);
      },
    },
  };
}
