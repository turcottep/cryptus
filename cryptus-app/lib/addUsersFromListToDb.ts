import CreateAccountFromWalletAddress from "./createAccountFromWalletAddress";

export default async function AddUsersToDb() {
  for (let i = 0; i < usersList.length; i++) {
    await CreateAccountFromWalletAddress(
      usersList[i].wallet_adress.toLocaleLowerCase(),
      true,
      usersList[i].username.toLocaleLowerCase()
    );
  }
}

export const usersList = [
  {
    username: "Seedphrase",
    wallet_adress: "0x1da5331994e781ab0e2af9f85bfce2037a514170",
  },
  {
    username: "Garyvee",
    wallet_adress: "0xd6a984153acb6c9e2d788f08c2465a1358bb89a7",
  },
  {
    username: "Beeple",
    wallet_adress: "0xc6b0562605d35ee710138402b878ffe6f2e23807",
  },
  {
    username: "flur",
    wallet_adress: "0xb32b4350c25141e779d392c1dbe857b62b60b4c9",
  },
  {
    username: "Pranksy",
    wallet_adress: "0xd387a6e4e84a6c86bd90c158c6028a58cc8ac459",
  },
  {
    username: "crypto888crypto",
    wallet_adress: "0x507fc313f9d50ce32ff94e67df48f7b0935d016a",
  },
  {
    username: "BeanieMaxi",
    wallet_adress: "0xabf107de3e01c7c257e64e0a18d60a733aad395d",
  },
  {
    username: "Artchick",
    wallet_adress: "0x0b8f4c4e7626a91460dac057eb43e0de59d5b44f",
  },
  {
    username: "fvckrender",
    wallet_adress: "0xfded90a3b1348425577688866f798f94d77a0d02",
  },
  {
    username: "fazebanks",
    wallet_adress: "0x7d4823262bd2c6e4fa78872f2587dda2a65828ed",
  },
  {
    username: "KeyboardMonkey",
    wallet_adress: "0xe1d29d0a39962a9a8d2a297ebe82e166f8b8ec18",
  },
  {
    username: "gmoneyNFT",
    wallet_adress: "0xf0d6999725115e3ead3d927eb3329d63afaec09b",
  },
  {
    username: "tropoFarmer ",
    wallet_adress: "0xa442ddf27063320789b59a8fdca5b849cd2cdeac",
  },
  {
    username: "Zeneca_33",
    wallet_adress: "0x886478d3cf9581b624cb35b5446693fc8a58b787",
  },
  {
    username: "VincentVanDough",
    wallet_adress: "0x0f0eae91990140c560d4156db4f00c854dc8f09e",
  },
  {
    username: "J1mmy",
    wallet_adress: "0x8ad272ac86c6c88683d9a60eb8ed57e6c304bb0c",
  },
  {
    username: "NateAlex",
    wallet_adress: "0x63a9dbce75413036b2b778e670aabd4493aaf9f3",
  },
  {
    username: "StephCurry",
    wallet_adress: "0x3becf83939f34311b6bee143197872d877501b11",
  },
  {
    username: "andy8052",
    wallet_adress: "0x90e5aa59a9df2add394df81521dbbed5f3c4a1a3",
  },
  {
    username: "PixelVault",
    wallet_adress: "0xfd64b63d4a54e6b1a0aa88e6623046c54f960d00",
  },
  {
    username: "valueandtime",
    wallet_adress: "0x82b47799859e354eac6dba1d0f6dd5b45124c14b",
  },
  {
    username: "mevcollcetor",
    wallet_adress: "0x5338035c008ea8c4b850052bc8dad6a33dc2206c",
  },
  {
    username: "VonMises14",
    wallet_adress: "0x8dbbca57ea56290efa14d835bbfd34faf1d89753",
  },
  {
    username: "Shamdoo",
    wallet_adress: "0x11360f0c5552443b33720a44408aba01a809905e",
  },
  {
    username: "Gfunkera86",
    wallet_adress: "0x115ab9e1dbe84030719835dd3d4b74503be8921b",
  },
  {
    username: "VitalikButerin",
    wallet_adress: "0xd8da6bf26964af9d7eed9e03e53415d37aa96045 ",
  },
  {
    username: "dan_opensea",
    wallet_adress: "0x530cf036ed4fa58f7301a9c788c9806624cefd19 ",
  },
];
