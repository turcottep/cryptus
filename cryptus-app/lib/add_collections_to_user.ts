import UsernameSetting from "../components/basic/settings/username_setting/username_setting";
import findAllUsers from "../pages/api/users/findAllUsers";

export default async function add_collections_to_user(
  user_collections_addresses: string[],
  user_collections_slugs: string[],
  username
) {
  console.log("Adding Col");
  const res = await fetch("/api/users/addCollectionsAddress", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      addresses: user_collections_addresses,
      username: username,
    }),
  });
  const user = await res.json();
  return user;
}
