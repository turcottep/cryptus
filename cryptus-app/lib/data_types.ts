export interface nft {
  [x: string]: any;
  properties: any;
  rarity_rank: number;
  image_url: string;
  external_url: string;
  name: string;
  last_sale_price: number;
  last_sale_symbol: string;
  collection: string;
  collection_slug: string;
  collection_size: number;
  collection_address: string;
  description: string;
  value?: number;
  token_id: string;
}

export interface dbUsers {
  id: string;
  email: string;
  username: string;
  displayName: string;
  nfts_order: Number[];
  description: string;
  views: Number;
  likes: Number;
  hash: string;
  collections_filter: string[];
  collections_list: string[];
  networth: Number;
  profile_image_url: string;
  subscription: string;
}

export interface user {
  networth?: number;
  networth_history?: number[];
  description?: string;
  username: string;
  address: string;
  collections_filter: string[];
  collections_list: string[];
  collections_address_list: string[];
  profile_image_url: string;
}

export interface nft_collection {
  id: number;
  name: string;
  address: string;
  description: string;
  image_url: string;
  external_url: string;
  value?: number;
  market_cap: number;
  nfts: nft[];
}

export interface profile_props {
  user: user;
  collections: nft_collection[];
}

export type collection = {
  id: string;
  logo: string;
  ticker: string;
  name: string;
  address: string;
  floor_price: number;
  floor_price_delta: number;
  user_owned: boolean;
  data_price: number[];
  data_volume: number[];
  timestamp: string;
};

export enum tabs {
  market = "market",
  research = "research",
  profile = "profile",
}

export enum intervals {
  week = 0,
  month = 1,
  three_months = 2,
  year = 3,
}

export const interval_days = [7, 30, 90, 365];
