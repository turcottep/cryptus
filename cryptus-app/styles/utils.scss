@use "/styles/variables" as v;

.cooltext {
  background-color: #38ef7d;
  background-image: linear-gradient(to right, #38ef7d, #11998e);
  background-image: conic-gradient(to right, #38ef7d, #11998e);

  background-size: 100%;
  background-clip: text;
  color: transparent;
  font-weight: bold;
  background-position: 0% 50%;
  text-shadow: 0px 0px 16px rgba(0, 0, 0, 0.2);
}

.cooltextgreen {
  background-color: #38ef7d;
  background-image: linear-gradient(to right, #38ef7d, #11998e);
  background-image: conic-gradient(to right, #38ef7d, #11998e);
  background-size: 100%;
  background-clip: text;
  color: transparent;
}

.cooltextred {
  background-color: #ea1b30;
  background-image: linear-gradient(to right, #ea1b30, #d01b5a);
  background-image: conic-gradient(to right, #ea1b30, #d01b5a);

  background-size: 100%;
  background-clip: text;
  color: transparent;
}

.cooltext_animated {
  width: 100%;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  background-size: 300% 300%;
  background-image: linear-gradient(
    -45deg,
    #38ef7d 0%,
    #11998e 25%,
    rgba(152, 68, 183, 1) 51%,
    rgba(255, 53, 127, 1) 100%
  );
  animation: AnimateBG 5s ease infinite;
  // overflow: hidden;
  // font-size: 250px;
  font-weight: bold;

  background-clip: text;
  color: transparent;
}

@keyframes AnimateBG {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

.coolcolor {
  color: #38ef7d;
  color: linear-gradient(to right, #38ef7d, #11998e);
  color: conic-gradient(to right, #38ef7d, #11998e);
}

.coolbutton {
  .button_disconnected {
    max-height: 32px;
    max-width: 140px;

    position: relative;
    overflow: hidden;

    // for background color shift
    background-image: linear-gradient(to right, #f27121, #8a2387);

    // to ease the button growth on hover
    transition: 0.6s;

    // psuedo-element shine animation left side
    &::before {
      content: "";
      display: block;
      position: absolute;
      background: rgba(255, 255, 255, 0.5);
      width: 60px;
      height: 100%;
      top: 0;
      filter: blur(30px);
      transform: translateX(300px) skewX(-15deg);
      user-select: none;
      pointer-events: none;
    }

    // psuedo-element shine animation right side
    &::after {
      content: "";
      display: block;
      position: absolute;
      background: rgba(255, 255, 255, 0.2);
      width: 30px;
      height: 100%;
      top: 0;
      filter: blur(5px);
      transform: translateX(300px) skewX(-15deg);

      user-select: none;
      pointer-events: none;
    }

    // grow button and change background gradient on hover
    &:hover {
      background-image: linear-gradient(to right, #000, #000);
      transform: perspective(100px) rotateX(-1deg) rotateY(2deg) scale(1.2);
      box-shadow: -5px -2px 20px -2px rgba(#f27121, 0.5);
      cursor: pointer;

      // psuedo-elements for right- and left-side shine animations
      &::before,
      &::after {
        transform: translateX(-300px) skewX(-15deg);
        transition: 0.7s;
      }
    }
  }
}
