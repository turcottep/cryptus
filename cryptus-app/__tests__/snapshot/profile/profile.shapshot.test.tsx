// components/__tests__/button.snapshot.test.tsx
import React from "react";
import renderer from "react-test-renderer";
import { profile_props, nft_collection, nft } from "../../../lib/data_types";

import { mockNextUseRouter } from "../../../utils/test_util";
import Profile from "../../../components/profile/profile";
import { mock_collection } from "../../../lib/mocks";
// import Props from "./mock_profile";

describe("<Profile />", () => {
  mockNextUseRouter({
    route: "",
    pathname: "",
    query: { userId: "test" },
    asPath: ``,
  });

  it("Profile renders correctly", () => {
    const collections: nft_collection[] = [mock_collection];

    const user = {
      networth: 420,
      description: "Here is a description",
      username: "A username",
      address: "An address",
    };
    const old_user = {
      id: "710e552b-2881-42de-afc5-4fd4d9ea698f",
      email: "turcotte@usherbrooke.ca",
      username: "turcotte",
      displayName: "Turcotte",
      nfts_order: [],
      description: "Je vais a MTL pour chercher un bike",
      views: 6000,
      likes: 400,
      hash: "4d6da51cf568dd8037650a5ae0fe1c4344cca65361f3096d4b6a926614ab59bc",
      address: "0x0da2f3401296427d302326cdf208b79f83abc995",
      wallets: [
        {
          id: "2df3578a-21fe-451f-9be3-065749899539",
          blockchain_id: "ETH",
          address: "0x0da2f3401296427d302326cdf208b79f83abc995",
          external_url:
            "https://api.opensea.io/api/v1/assets?owner=0x0da2f3401296427d302326cdf208b79f83abc995&order_direction=asc&offset=0&limit=50",
          userId: "710e552b-2881-42de-afc5-4fd4d9ea698f",
        },
      ],
    };

    const tree = renderer
      .create(
        <Profile collections={collections} user={user} isMobile={false} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
