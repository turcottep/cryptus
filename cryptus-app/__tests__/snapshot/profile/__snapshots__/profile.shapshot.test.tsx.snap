// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> Profile renders correctly 1`] = `
<div>
  <div>
    <div>
      <a
        onClick={[Function]}
        target="_blank"
      >
        <img
          src="/icons/back_icon.png"
        />
      </a>
      <div>
        test
      </div>
      <div
        onClick={[Function]}
      >
        <img
          src="icons/menu_icon.png"
        />
      </div>
    </div>
    <div>
      <div>
        <img
          alt="Avatar"
          src="./icons/icon-192x192.png"
        />
      </div>
      <div>
        <div>
          turcotte
        </div>
      </div>
      <div>
        Je vais a MTL pour chercher un bike
      </div>
      <div>
        NETWORTH:undefined eth
      </div>
    </div>
    <div>
      <div>
        <div>
          <div
            onClick={[Function]}
          >
            <div />
          </div>
          <div>
            CryptoKitties
          </div>
          <div>
            
          </div>
        </div>
      </div>
    </div>
    <div
      id="footer"
    >
      <a
        href="/market"
      >
        <img
          src="/icons/market_overview_icon.png"
        />
      </a>
      <a>
        <img
          src="/icons/research_icon.png"
        />
      </a>
      <a>
        <img
          src="/icons/notification_icon.png"
        />
      </a>
      <a
        href="/"
      >
        <img
          src="/icons/profile_icon.png"
        />
      </a>
    </div>
  </div>
</div>
`;
